Hunter package manager support and some fixes

themarpe opened the pull request:
Hi.
This PR adds a couple of changes.

- Hunter package manager. It is dependent only on CMake. Right now it's disabled by default, but can be enabled using `-DHUNTER_ENABLED=ON`.
Using Hunter, there is no need to have Nlohmann JSON target available.

- Fix which was needed to make install work under Hunter and I think also fixes issue #77 

- MSVC compilation error fix (src/json-validator.cpp)

I'm also interested in adding this package to Hunter - this will enable users to use this library with ease

pboettch commented:
I'll add you changes to the master-branch. It would be nice to have a build-test using hunter in the test-dir. Could you provide one?

themarpe commented:
Should I create another PR or can this one be reopened?

I've added additional commits to pr-hunter branch, which fixes differences that arose with bringing over the code to master.
 - Hunter should be added before project command
 - Added tests (.travis.yml) which execute all tests but with json.hpp retrieved with Hunter package manager

Also two questions, why bring over the changes manually without merging and was the MSVC fix left out intentionally (json-validator.cpp)?

pboettch commented:
> Should I create another PR or can this one be reopened?

A new one.

> I've added additional commits to pr-hunter branch, which fixes differences that arose with bringing over the code to master.
> 
>     * Hunter should be added before project command
> 
>     * Added tests (.travis.yml) which execute all tests but with json.hpp retrieved with Hunter package manager

Great! Rebase on the project's master-branch and we'll discuss/integrate.

> Also two questions, why bring over the changes manually without merging and was the MSVC fix left out intentionally (json-validator.cpp)?

I prefer rebase over merging and due to your unrelated MSVC-fix (which I totally forgot about, sorry) I applied the changes manually.

That said: please redo a new PR for the MSVC-fix as well.
